File: DSA with python/BinarySearch.py
li = list(map(int, input("Enter numbers separated by space: ").split()))
li.sort()
print("Sorted list:", li)

target = int(input("Enter the number to search: "))

start = 0
end = len(li) - 1
found = -1   # flag for position

count = 0
for i in range(0, end):
    if li[i]==target:
        count +=1


while(start<=end):
    mid = (start + end) // 2
    if li[mid] == target:
        found = mid
        break
    elif li[mid] < target:
        start = mid + 1
    else:
        end = mid - 1

if found != -1:
    print("The first found position is", found + 1, "and total count of target is", count)  # +1 for human-readable index
else:
    print("The target is not found")

File: DSA with C/searching/binarySearch.c
// Binary search:


#include<stdio.h>
int main(){
    int n, target, flag = 0, index, start, mid, end;
    printf("Enter the array size: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter the array: ");

    for(int i=0; i<n; i++){
        scanf("%d", &arr[i]);
    }
    // for(int i=0; i<n; i++){
    //     printf("%d ", arr[i]);
    // }
    printf("Enter the target: ");
    scanf("%d", &target);

    start = 0;
    end = n-1;

    while(start<=end){
        mid = (start+end)/2;
        if(arr[mid]==target){
            flag = 1;
            index = mid;
            break;
        }
        else if(arr[mid]> target){
            end = mid - 1;
        }
        else{
            start = mid +1;
        }
    }
    if(flag) printf("The target is found at the position: %d", index+1);
    else printf("The target is not found.");
    return 0;

}